internal/controller/storageclient_controller.go

```diff
@@ -309,26 +401,155 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err != nil {
 				return reconcile.Result{}, fmt.Errorf("failed to create remote noobaa: %v", err)
 			}
-		}
-	}
-	if r.storageClient.GetAnnotations()[storageClaimProcessedAnnotationKey] != "true" {
-		if err := r.reconcileBlockStorageClaim(); err != nil {
-			return reconcile.Result{}, err
-		}
-
-		if err := r.reconcileSharedfileStorageClaim(); err != nil {
-			return reconcile.Result{}, err
-		}
+		case "StorageClass":
+			storageClass := &storagev1.StorageClass{}
+			if err := json.Unmarshal(eResource.Data, &storageClass); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal storage configuration response: %v", err)
+			}
+			if err := r.own(storageClass); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to own Storage Class resource: %v", err)
+			}
+			utils.AddLabels(storageClass, eResource.Labels)
```

Comment:
Had a silly question, CreateOrReplace will fetch the resource, apply the mutate function, and then compare the fetched resource with mutated right? 
If we don't add labels under CreateOrReplace won't that mean that labels won't persist?
Similarly won't we have the same problem for owning the StorageClass

Reply:
done, another reason why I didn't do is, initial assumption was storageclass will be reconciled at ocs-op as well and now that isn't correct made the changes rather than waiting for another release.